tests/testthat/test-trailing_blank_lines_linter.R

```diff
@@ -17,22 +17,39 @@ test_that("returns the correct linting", {
 
   # Construct a test file without terminal newline
   # cf. test-get_source_expressions.R
-  writeLines("lm(y ~ x)", tmp <- tempfile())
-  on.exit(unlink(tmp), add = TRUE)
-  writeBin(
-    # strip the last (two) element(s) (\r\n or \n)
-    head(
-      readBin(tmp, raw(), file.size(tmp)),
-      if (.Platform$OS.type == "windows") -2L else -1L
-    ),
-    tmp2 <- tempfile()
-  )
-  on.exit(unlink(tmp2), add = TRUE)
+  tmp <- withr::local_tempfile()
+  tmp2 <- withr::local_tempfile()
+  cat("lm(y ~ x)\n", file = tmp)
+  cat("lm(y ~ x)", file = tmp2)
 
   expect_lint(content = NULL, file = tmp, NULL, linter)
   expect_lint(content = NULL, file = tmp2, list(
     message = msg2,
     line_number = 1L,
     column_number = 10L
   ), linter)
+
+  # Construct an Rmd file without terminal newline
+  tmp3 <- withr::local_tempfile()
+  cat(
+    trim_some(
+      '---
+      title: "Some file"
+      ---
+
+      ```{r}
+      abc = 123
+      ```
+
+      ```{r child="some-file.Rmd"}
```

Comment:
false positive lint here?

Reply:
also note the NA -- seems the linter name has not been picked up correctly

Reply:
It seems the test code is linted as Rmd for some reason?!
The NA should be investigated in a separate issue. Maybe we stumbled on an actual bug in the extractor code.

Reply:
Can't seem to find the issue from a quick search but AFAIK that should be related to the known issue where our logic for classifying Rmd is not the best.

Reply:
Filed #1405 for the NA issue